integration-libs/opf/base/core/facade/opf-payment.service.ts

```diff
@@ -0,0 +1,155 @@
+      payload.paymentSessionId,
+      payload.paymentVerificationPayload
+    )
+  );
+
+  protected submitPaymentCommand: Command<
+    {
+      submitInput: SubmitInput;
+    },
+    boolean
+  > = this.commandService.create((payload) => {
+    return this.opfPaymentHostedFieldsService.submitPayment(
+      payload.submitInput
+    );
+  });
+
+  protected submitCompletePaymentCommand: Command<
+    {
+      submitCompleteInput: SubmitCompleteInput;
+    },
+    boolean
+  > = this.commandService.create((payload) => {
+    return this.opfPaymentHostedFieldsService.submitCompletePayment(
+      payload.submitCompleteInput
+    );
+  });
+
+  protected afterRedirectScriptsCommand: Command<
+    {
+      paymentSessionId: string;
+    },
+    AfterRedirectScriptResponse
+  > = this.commandService.create((payload) => {
+    return this.opfPaymentConnector.afterRedirectScripts(
+      payload.paymentSessionId
+    );
+  });
+
+  protected ctaScriptsCommand: Command<
+    {
+      ctaScriptsRequest: CtaScriptsRequest;
+    },
+    CtaScriptsResponse
+  > = this.commandService.create((payload) => {
+    return this.opfPaymentConnector.getCtaScripts(payload.ctaScriptsRequest);
+  });
+
+  protected applePaySessionCommand: Command<
+    {
+      applePayWebSessionRequest: ApplePaySessionVerificationRequest;
+    },
+    ApplePaySessionVerificationResponse
+  > = this.commandService.create((payload) => {
+    return this.opfPaymentApplePayService.getApplePayWebSession(
+      payload.applePayWebSessionRequest
+    );
+  });
+
+  protected activeConfigurationsQuery: Query<ActiveConfiguration[]> =
+    this.queryService.create<ActiveConfiguration[]>(() =>
+      this.opfPaymentConnector.getActiveConfigurations()
+    );
+
+  constructor(
+    protected queryService: QueryService,
+    protected commandService: CommandService,
+    protected opfPaymentConnector: OpfPaymentConnector,
+    protected opfPaymentHostedFieldsService: OpfPaymentHostedFieldsService,
+    protected opfPaymentApplePayService: OpfPaymentApplePayService
+  ) {}
+
+  verifyPayment(
+    paymentSessionId: string,
+    paymentVerificationPayload: OpfPaymentVerificationPayload
+  ): Observable<OpfPaymentVerificationResponse> {
+    return this.verifyPaymentCommand.execute({
+      paymentSessionId,
+      paymentVerificationPayload,
+    });
+  }
+
+  submitPayment(submitInput: SubmitInput): Observable<boolean> {
+    return this.submitPaymentCommand.execute({
+      submitInput,
+    });
+  }
+
+  submitCompletePayment(
+    submitCompleteInput: SubmitCompleteInput
+  ): Observable<boolean> {
+    return this.submitCompletePaymentCommand.execute({ submitCompleteInput });
+  }
+
+  afterRedirectScripts(paymentSessionId: string) {
+    return this.afterRedirectScriptsCommand.execute({ paymentSessionId });
+  }
+
+  getActiveConfigurationsState(): Observable<
+    QueryState<ActiveConfiguration[] | undefined>
+  > {
+    return this.activeConfigurationsQuery.getState();
+  }
```

Comment:
[QUESTION]

Does really Opf `ActiveConfiguration` refer only to Payment? (it's now in Opf Payment Service)

I can see it the code it's used also for CTA, which I guess it a totally independent feature from Payments.
So shall we move the `getActiveConfigurationState` out from `OpfPaymentServie`, `OpfPaymentConnector` and `OpfPaymentAdapter` to some other service? e.g. to the main `OpfService`?